crates/bevy_render/src/view/window.rs

```diff
@@ -172,25 +180,28 @@ pub fn prepare_windows(
         .filter(|x| x.raw_handle.is_some())
     {
         let window_surfaces = window_surfaces.deref_mut();
-        let surface = window_surfaces
+        let surface_data = window_surfaces
             .surfaces
             .entry(window.id)
             .or_insert_with(|| unsafe {
                 // NOTE: On some OSes this MUST be called from the main thread.
-                render_instance.create_surface(&window.raw_handle.as_ref().unwrap().get_handle())
+                let surface = render_instance
+                    .create_surface(&window.raw_handle.as_ref().unwrap().get_handle());
+                let format = *surface
+                    .get_supported_formats(&render_adapter)
+                    .get(0)
+                    .unwrap_or_else(|| {
+                        panic!(
+                            "No supported formats found for surface {:?} on adapter {:?}",
+                            surface, render_adapter
+                        )
+                    });
+                SurfaceData { surface, format }
             });
 
         // Creates a closure to avoid calling this logic unnecessarily
```

Comment:
the closure trick is removed by this PR. Should it be kept? Or the comment removed?

Reply:
Comment should be removed. On it!